File: Solutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res=0;
        unordered_map<char,int> m;
        for(int i=0,j=0;i<s.size();++i){
            if(m.count(s[i]))
                j=max(j,m[s[i]]);
            m[s[i]]=i+1;
            res=max(res,i-j+1);
        }
        return res;
    }
};

File: Solutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        m={}
        i,j,res=0,0,0
        while j<len(s):
            if s[j] in m:
                i=max(i,m[s[j]]+1)
            m[s[j]]=j
            res=max(res,j-i+1)
            j+=1
        return res